app/controllers/invites_controller.rb

```diff
@@ -109,73 +78,104 @@ def create
     end
   end
 
-  def create_invite_link
-    params.permit(:email, :max_redemptions_allowed, :expires_at, :group_ids, :group_names, :topic_id)
+  def update
+    invite = Invite.find_by(invited_by: current_user, id: params[:id])
+    raise Discourse::InvalidParameters.new(:id) if invite.blank?
 
-    is_single_invite = params[:email].present?
-    unless is_single_invite
-      guardian.ensure_can_send_invite_links!(current_user)
+    if params[:topic_id].present?
+      topic = Topic.find_by(id: params[:topic_id])
+      guardian.ensure_can_invite_to!(topic)
     end
 
-    groups = Group.lookup_groups(
-      group_ids: params[:group_ids],
-      group_names: params[:group_names]
-    )
-    if !guardian.can_invite_to_forum?(groups)
-      raise StandardError.new I18n.t("invite.cant_invite_to_group")
+    if params[:group_ids].present? || params[:group_names].present?
+      groups = Group.lookup_groups(group_ids: params[:group_ids])
+      guardian.ensure_can_invite_to_forum!(groups)
     end
-    group_ids = groups.map(&:id)
 
-    if is_single_invite
-      invite_exists = Invite.exists?(email: params[:email], invited_by_id: current_user.id)
-      if invite_exists && !guardian.can_send_multiple_invites?(current_user)
-        return render json: failed_json, status: 422
-      end
-
-      if params[:topic_id].present?
-        topic = Topic.find_by(id: params[:topic_id])
+    Invite.transaction do
+      invite.topic_invites.destroy_all
+      invite.topic_invites.create!(topic_id: topic.id) if topic.present?
 
-        if topic.present?
-          guardian.ensure_can_invite_to!(topic)
-        else
-          raise Discourse::InvalidParameters.new(:topic_id)
-        end
-      end
-    end
+      invite.invited_groups.destroy_all
+      groups.each { |group| invite.invited_groups.find_or_create_by!(group_id: group.id) } if groups.present?
 
-    invite_link = if is_single_invite
-      Invite.generate_single_use_invite_link(params[:email], current_user, topic, group_ids)
-    else
-      Invite.generate_multiple_use_invite_link(
-        invited_by: current_user,
+      invite.update!(
         max_redemptions_allowed: params[:max_redemptions_allowed],
-        expires_at: params[:expires_at],
-        group_ids: group_ids
+        expires_at: params[:expires_at]
       )
     end
-    if invite_link.present?
-      render_json_dump(invite_link)
-    else
-      render json: failed_json, status: 422
-    end
-  rescue => e
-    render json: { errors: [e.message] }, status: 422
+
+    render json: success_json
   end
 
   def destroy
     params.require(:id)
 
     invite = Invite.find_by(invited_by_id: current_user.id, id: params[:id])
     raise Discourse::InvalidParameters.new(:id) if invite.blank?
+
     invite.trash!(current_user)
 
     render json: success_json
   end
 
+  def perform_accept_invitation
+    params.require(:id)
```

Comment:
Minor but I think we should change `:id` to `:invite_key` instead so that it isn't confusing the two columns.